Marlin/src/inc/Conditionals_LCD.h

```diff
@@ -381,6 +381,9 @@
 #define HAS_LCD_MENU        (ENABLED(ULTIPANEL) && DISABLED(NO_LCD_MENUS))
 #define HAS_ADC_BUTTONS      ENABLED(ADC_KEYPAD)
 
+// Does the Extensible UI provide a pin mapping for RepRapDiscount connectors?
+#define HAS_NEWPANEL_PINMAP ENABLED(LULZBOT_TOUCH_UI) && ANY(AO_EXP1_PINMAP, AO_EXP2_PINMAP, CR10_TFT_PINMAP)
```

Comment:
Note that `NEWPANEL` refers to a display having EN1, EN2, and ENC pins for a rotary encoder. The Ultimaker ULTIPANEL is the archetype followed by the RRD displays, and we assume that `ULTIPANEL` will always have a `NEWPANEL` encoder. The RRW keypad, though not an LCD, also has a `NEWPANEL` rotary encoder.

If the Lulzbot Touch UI only needs the encoder, but doesn't care about the other pins for the display, then the `NEWPANEL` block in the pins files can be moved outside of the `HAS_SPI_LCD` block, and skip defining the other pins that are only needed for a real SPI-based controller.

Reply:
@thinkyhead: The intent for `HAS_NEWPANEL_PINMAP` is that it requests that the following constants be defined even if `ULTIPANEL` or `NEWPANEL` are not actually enabled:

- BEEPER_PIN
- BTN_EN1
- BTN_EN2
- BTN_ENC
- LCD_PINS_RS
- LCD_PINS_ENABLE
- LCD_PINS_D4
- LCD_PINS_D5
- LCD_PINS_D6
- LCD_PINS_D7
- SD_DETECT_PIN

These corresponds to GPIO lines that are available on the 10-pin EXP1 and EXP2 connectors on boards that support the [RepRapDiscount Full Graphic Smart Controller](https://reprap.org/wiki/RepRapDiscount_Full_Graphic_Smart_Controller). The Lulzbot Extensible UI then re-uses those pins to communicate with the display module.

The intent behind this is to make the Lulzbot display compatible with the connector on all boards that already support the RepRapDiscount Full Graphic Smart Controller display using a 10-wire ribbon cable.

 I do not understand the nuances between `NEWPANEL`, `ULTRALCD`  etc, so I only guessed that `HAS_NEWPANEL_PINMAP` was an appropriate name. Feel free to change that if you have something more appropriate.

The intended meaning of `HAS_NEWPANEL_PINMAP` is to force the definitions of the pin constants as if a RepRapDiscount Full Graphics Smart Controller was available, but *do not* enable the code in Marlin to actually drive that display.

Reply:
I noticed you renamed `HAS_NEWPANEL_PINMAP` as `TOUCH_UI_ENCODER`.

This is not correct. The touch panel never has an encoder. However, the BTN_EN1, BTN_EN2 and BTN_EN may be used for signalling to the display. Take a look at this file and maybe things will be a bit clearer:

https://github.com/MarlinFirmware/Marlin/pull/14967/files#diff-20197638d22619bad0559e91cd1cb358

To be honest, the whole pin mapping thing is not absolutely necessary. We could have people manually select the pins for their board by defining `CLCD_MOD_RESET`, `CLCD_SPI_CS`, `CLCD_SOFT_SPI_MOSI`, `CLCD_SOFT_SPI_MISO` and `CLCD_SOFT_SPI_SCLK`, but the pin mapping allows a PCB to be made which has a 10-pin connector that can be used on any printer as a drop in replacement for the RepRapDiscount Full Graphic Smart Controller without having to define any new pins at all.

Reply:
Ah, I see. You are letting the pins be defined as if they serve one function, but in fact they serve other functions. It would be better if we can tell at a glance within the pins files exactly what the functions of the pins are going to be. Of course, the need for this hackery stems from a general weakness of the LCD pin mapping, in that we must define the LCD pin mappings separately for each board, and there's no intermediate step (at this time) that would allow us to define the pins for each type of LCD just once.